### DISCUSSION TITLE
Creality Halot One

### DISCUSSION THREAD

**Discussion by xPew** [sn4k3/UVtools#391] [Q&A] [answered]
Hi!

I am thinking about a Creality Halot One resin printer. Haven't found much info about if that could be used with UVtools and .ctb format/smart features like dynamic layers, variable exposure... The LD002h got an update that enables nearly all feauters I can think of, but the newer Halot One has a self developed Creality mainboard, not a chitu one (its not that bad since chitu closed down its mainboards...), but I don't know what does this mean in terms of features.
Also it has its own slicing software, but according to the changelog here, I can still use PrusaSlicer and convert the file for the Halot one?
Anyone has experience with it, or can tell me what can I expect out of it? Creality cxdlp what I am looking for right?

Best regards

**Comment by frankvdh** [comment]
I've just bought a Halot-one and discovered UVTools. I'm just beginning to claw my way up the learning curve, so haven't a lot to share yet. I wasn't aware that there were different flavours of .cxdlp. So just a couple of data points...

UVTools won't even read a .cxdlp file created by Chitubox for the Halot One. 
```
System.IO.FileLoadException: Not a valid CXDLP file!
   at UVtools.Core.FileFormats.CXDLPFile.Footer.Validate() in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.Core\FileFormats\CXDLPFile.cs:line 389
   at UVtools.Core.FileFormats.CXDLPFile.DecodeInternally(OperationProgress progress) in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.Core\FileFormats\CXDLPFile.cs:line 946
   at UVtools.Core.FileFormats.FileFormat.Decode(String fileFullPath, FileDecodeType fileDecodeType, OperationProgress progress) in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.Core\FileFormats\FileFormat.cs:line 3291
   at UVtools.WPF.MainWindow.<>c__DisplayClass103_0.<ProcessFile>b__1() in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.WPF\MainWindow.axaml.cs:line 1420
```
When I use UVTools to validate .cxdlp files created by Halot-box and Lychee, it complains (correctly, I think... the height should surely be 80mm?) about the printer image ratio: 
```
It looks like this file was sliced with an incorrect image ratio.
Printing this file may produce a stretched model with wrong proportions or a failed print.
Please go back to Slicer configuration and validate the printer resolution and display size.

Resolution: {Width=1620, Height=2560}
Display: {Width=128, Height=81}
Ratio: 20:1
```
The printer wouldn't open files saved by UVTools (I wasn't aware that the "Convert To Creality CXDLP" capability existed -- thanks for that!). I'll try loading files converted to Creality v2 and v3 when the current print finishes.

I'm only aware of Chitubox, Halot-box, and Lychee. I'd be interested to know what other slicers can be used, how good they are, and how to convert their output to Creality .cxdlp format. The only slicers I know of only output G-code.

Also, there's an interesting blog on hacking the Halot One at https://www.creationfactory.co/2022/01/rooting-creality-halot-one-resin-3d.html.

**Reply by sn4k3** [reply]
> System.IO.FileLoadException: Not a valid CXDLP file!

Either you are using an not cxdlp file or using the first flavor of cxdlp (v1) which is not supported by UVtools anymore, if you use v1 please consider updating your firmware and start using v3.

> It looks like this file was sliced with an incorrect image ratio.
> Resolution: {Width=1620, Height=2560}
> Display: {Width=128, Height=81}
> Ratio: 20:1

If you carefull look at message error, that is a real problem, you have **1620px** to **128mm** of display and **2560px** to **81mm** of display. Meaning display values are flip! Look again:

```
 px     mm
1620 - 128
2560 - 81
```

So the shortest resolution (1620=X) have the longest display side (128mm)?
And the longest resolution (2560=Y) have the shortest display side (81mm)?
This is just wrong! It looks like you did your own printer profile instead of pick up a preset. Because lychee presets are correct and do not look like what you posted

![LycheeSlicer_2022-03-29_00-38-32](https://user-images.githubusercontent.com/113281/160504572-666be92b-0d16-4931-aed8-ae5a8a55a18a.png)

> I'm only aware of Chitubox, Halot-box, and Lychee. I'd be interested to know what other slicers can be used, how good they are, and how to convert their output to Creality .cxdlp format. The only slicers I know of only output G-code.

You also have PrusaSlicer + UVtools and VoxelDance Tango.

> Also, there's an interesting blog on hacking the Halot One

Halot series are linux machines, which means they are like linux PCs, so you can do whatever you want, including creating SAMBA network, webservers, etc

**Reply by frankvdh** [reply]
> > System.IO.FileLoadException: Not a valid CXDLP file!
> 
> Either you are using an not cxdlp file or using the first flavor of cxdlp (v1) which is not supported by UVtools anymore, if you use v1 please consider updating your firmware and start using v3.

Thanks for your knowledgeable reply. I'm completely new to resin printing, so it's entirely possible I screwed up, but I didn't **try** to do anything other than use the pre-defined and default settings. Anyway I checked that I have the latest version of Chitubox (v1.9.2) and saved a file from there and got the same result. Who knows, maybe Chitubox writes v1 cxdlp files. It's all moot because I think I prefer the PrusaSlicer/UVTools toolchain.

LycheeSlicer is already uninstalled... its 30 second delays are just so irritating that I can't even be bothered to try it for free.

I will definitely update my firmware when I figure out how to do that! So much to learn....

**Comment by wjsteele** [comment]
I use UVTools all the time on my Halot One, so I know it works.  Also, as for slicers, I prefer to use PrusaSlicer

**Reply by frankvdh** [reply]
OK...
For any newb who are more newb than me, and for a less-newb person to correct if I've got it wrong...
The setup process is not obvious:
1. Install **both** UVTools and PrusaSlicer.
2. In UVTools **Help** menu, choose **"Install profiles into PrusaSlicer"**
3. Select **"Creality Halot One CL-60"** from the right-hand "Printer Profiles" list
4. Maybe select some "Print Profiles" from the left-hand list (but you can always come back and do this later)
5. Click **"Install selected profiles"** button
6. In PrusaSlicer, you should now have a "Creality Halot One CL-60" printer defined in Printer Settings.

Slicing and printing:
1. Load STL files into PrusaSlicer, manipulate, them, etc. It appears that PrusaSlicer automatically lifts the model and adds supports underneath it, to facilitate removing it from the build plate. Nice!
2. Slice
3. Export -- this will create a **.sl1** file
4. In **UVTools**, load the **.sl1** file
5. UVTools will automatically convert it to **"Creality CXDLP" V2** (presumably it knows this is the right format because I did it once already, or because it knows I have a Halot One)
6. Copy the **.cxdlp** file to a USB stick
7. Plug the stick into your printer and select the file

[Edited based on sn4ke's reply]

**Reply by sn4k3** [reply]
Step 5) is unnecessary, it will auto convert to cxdlp once you load .sl1 in UVtools. Look at title bar and notice the file change extension to .cxdlp once open .sl1.

> V2 (V3 doesn't work)

This depend on each person printer firmware. And can be auto set on prusa slicer, under printer - Notes: `FILEVERSION_2` just after `FILEFORMAT_CXDLP`

**Comment by sn4k3** [comment, marked as answer]
It come a bit late, but answering @xPew :

> Also it has its own slicing software, but according to the changelog here, I can still use PrusaSlicer and convert the file for the Halot one?

Yes, UVtools allow use any slicer and convert to cxdlp.

> Anyone has experience with it, or can tell me what can I expect out of it? Creality cxdlp what I am looking for right?

Hardware-wise they are good machines, in terms of light and print
Firmware-wise is one of the worst and more limited formats, meaning you can't do much "hacking", no different settings per layer, which means you are stuck with global values. It also lacks of some fields.
It looks like making a good printer that marks every check is a impossible mission. And the most stupid thing is that they can fix firmware and make a proper thing, as that come in form of a virtual update. Still they doesn't... That would be one day of proper work.

In end of the day this will give you good prints but thats all, do not expect to use exotic features of UVtools or slicers innovations. This limiting factors on firmwares are a plague that i can't understand why they are so bad and limited
